Homework 10

Dragos-Andrei-75 opened the pull request:
Defined endpoints:
- findReservationsFromLibraryForPeriod
    - searches through the reservations made for exemplars of books from a specific library and returns a page containing sorted reservations made between two specified dates;
- findReservationsFromUserByStatus
    - searches through the reservations made by a specific user and returns a page containing sorted reservations with a specified status;

Dragos-Andrei-75 committed:
Homework 10

Defined endpoints:
- findReservationsFromLibraryForPeriod
    - searches through the reservations
      made for exemplars of books from
      a specific library and returns a
      page containing sorted
      reservations made between two
      specified dates;
- findReservationsFromUserByStatus
    - searches through the reservations
      made by a specific user and returns
      a page containing sorted
      reservations with a specified status;

marcopacurariu3 commented on src/main/java/com/example/SpringBookstore/repositories/ExemplaryRepository.java:
```diff
@@ -30,18 +30,18 @@ public interface ExemplaryRepository extends JpaRepository<Exemplary, Long> {
+            AND e.ID NOT IN (
+                SELECT r.EXEMPLARY_ID FROM reservations r
+                WHERE r.EXEMPLARY_ID = e.ID
+                AND NOT (:startDate > r.END_DATE OR :endDate < r.START_DATE)
```
sa nu uiti sa tii cont si de status-uri

marcopacurariu3 commented on src/main/java/com/example/SpringBookstore/repositories/UserRepository.java:
```diff
@@ -1,12 +1,30 @@
+                    WHERE r.USER_ID = :userID
+                    AND (r.STATUS = NULL OR r.STATUS = :reservationStatusName)
+                    """,
+            nativeQuery = true)
```
cum de a fost nevoie de native query aici?

marcopacurariu3 commented on src/main/java/com/example/SpringBookstore/controllers/ReservationController.java:
```diff
@@ -0,0 +1,67 @@
+    }
+
+    @GetMapping(path = "/library/{libraryID}")
+    public ResponseEntity<?> findReservationsFromLibraryForPeriod(@PathVariable(value = "libraryID") Long libraryID, @RequestParam(required = false) LocalDate startDate, @RequestParam(required = false) LocalDate endDate, @RequestParam(required = false) Integer pageNumber, @RequestParam(required = false) Integer pageSize, @RequestParam(required = false, defaultValue = "ASC") String sortDirection,  ...
```
esti sigur ca merge cu default value "START_DATE"? stiu ca trebuia sa se numeasca field-ul fix cum e si Java (nu in DB)

marcopacurariu3 commented:
looks good, nice job!

Dragos-Andrei-75 committed:
Addressed All Issues from the Previous Review

Addressed all issues from the previous review made by Marco Păcurariu.

Added the Bookstore.postman_collection.json file into the resources folder.

Bookstore.postman_collection.json contains request for testing each endpoint that exist within the project.
